sdk/core/azure-core/azure/core/pipeline/policies/_authentication_async.py

```diff
@@ -38,7 +38,20 @@ class AsyncBearerTokenCredentialPolicy(AsyncHTTPPolicy[HTTPRequestType, AsyncHTT
     def __init__(self, credential: "AsyncTokenCredential", *scopes: str, **kwargs: Any) -> None:
         super().__init__()
         self._credential = credential
-        self._lock = asyncio.Lock()
+
+        try:
+            # Check if we are running in an asyncio event loop.
+            asyncio.get_running_loop()
+            self._lock = asyncio.Lock()
+        except RuntimeError:
+            # Otherwise, assume we are running in a trio event loop.
+            try:
+                import trio  # pylint:disable=networking-import-outside-azure-core-transport
+
+                self._lock = trio.Lock()
+            except ImportError as err:
+                raise RuntimeError("AsyncBearerTokenCredentialPolicy requires an asyncio or trio event loop.") from err
```

Comment:
Customer may be confused. Maybe some thing like:

```suggestion
                raise RuntimeError("This code needs to run as asyncio.run or trio.run.") from err
```

I want to make it actionable by a user receiving the error, what they should do specifically. I'm afraid they won't even know what is the policy name in the first place.